src/models/orders.rs

```diff
@@ -0,0 +1,142 @@
+extern crate anyhow;
+extern crate ethabi;
+extern crate ethsign;
+extern crate rustc_hex;
+extern crate serde_json;
+extern crate web3;
+
+use anyhow::Result;
+use ethabi::encode;
+use ethsign::Signature;
+use serde::{Deserialize, Serialize};
+use std::cmp::Ord;
+use std::cmp::Ordering;
+use std::cmp::PartialOrd;
+use std::ops::Mul;
+use web3::contract::tokens::Tokenizable;
+use web3::signing::keccak256;
+use web3::types::{Address, H160, H256, U256};
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, Default)]
+pub struct Order {
+    pub sell_amount: U256,
+    pub buy_amount: U256,
+    pub current_sell_amount: U256,
+    pub current_buy_amount: U256,
+    pub buy_token: Address,
+    pub sell_token: Address,
+    pub owner: Address,
+    pub nonce: u8,
+    pub signature_v: u8,
+    pub signature_r: H256,
+    pub signature_s: H256,
+    pub valid_until: U256,
+}
+
+impl Order {
+    #[allow(dead_code)]
+    pub fn get_digest(&self) -> Result<[u8; 32]> {
+        let domain_separator: H256 = serde_json::from_str(
+            r#""0x24a654ed47680d6a76f087ec92b3a0f0fe4c9c82c26bff3bb22dffe0f120c7f0""#,
+        )
+        .unwrap();
```

Comment:
Just curious, where does this domain separator come from? Also, I would prefer the H256 `FromStr` implementation directly (that way you avoid the need to escape the `"` with the `r##` here-doc:
```suggestion
        let domain_separator: H256 = "0x24a654ed47680d6a76f087ec92b3a0f0fe4c9c82c26bff3bb22dffe0f120c7f0"
            .parse()
            .unwrap();
```

Reply:
Its coming from here: https://github.com/josojo/oba-contracts/blob/master/src/contracts/PreAMMBatcher.sol#L123
It's good practice to use them so that the orders valid for one domain/contract are not valid somewhere else.

Reply:
Cool, makes perfect sense. Learned something new :smile: